### ISSUE TITLE
Hide scrollbar when "Other interesting profiles" section is scrollable

### ISSUE THREAD

**Issue by wholesomedev** [hollowverse-archive/hollowverse#246]

**Comment by forabi** [comment]
We are setting:

```scss
overflow-x: overlay;
```

I thought this hides the scroll bar on touch devices?

**Comment by wholesomedev** [comment]
I was able to see the scroll bar on iPhone. Can you try it and see?

**Comment by forabi** [comment]
* Safari on iOS 11: Scrollbar is not visible initially, when I start scrolling, an "overlay" scrollbar becomes visible.
* Chrome on Android: same behavior as Safari on iOS
* Chrome on iOS: Scrollbar is never shown.

It's interesting to see Chrome and Safari behaving differently on iOS. I thought all browsers have to use the native Web View component on iOS?

**Comment by wholesomedev** [comment]
@forabi Yeah, that's what I thought too. I haven't tested in Chrome.

**Comment by forabi** [comment]
@wholesomedev So do we want to hide the overlay scrollbar that shows when the user starts scrolling?

**Comment by forabi** [comment]
> Chrome on iOS: Scrollbar is never shown.

Never mind this. I just tested again and now it shows 🤷‍♂️

**Comment by wholesomedev** [comment]
Yes I think we should hide the overlay scroll. The homepage of Google has
that behavior where they hide the overlay scroll bar.

**Comment by forabi** [comment]
I've used the `pointer: coarse` media query and `::webkit-scrollbar` to hide the scrollbar on touch devices only, as we don't want to break scrolling on desktops. This will land in #230.

**Comment by forabi** [comment]
Looks like this only works on Chrome for Android. It does not work on iOS, although the pseudo-element is prefixed with `::-webkit-` 😅

**Comment by wholesomedev** [comment]
Since Google.com has side-scrolling cards without a scrollbar, I looked into how they pulled that off.

They're using a very nasty hack. They add an empty div to the bottom of the scrolling container to artificially increase its height, and then they use `margin-bottom: -20px` to hide the scrolling bar.

I will downgrade this issue to high difficulty low impact.

**Comment by forabi** [comment]
> They're using a very nasty hack. They add an empty div to the bottom of the scrolling container to artificially increase its height, and then they use margin-bottom: -20px to hide the scrolling bar.

Whoa! I tried many solutions but none worked on Safari. Should we do the same here?

**Comment by wholesomedev** [comment]
>Should we do the same here?

If we're going to implement it, I think we should abstract this behavior in its own component, so that we can hide the scroll bar of any side-scrolling UI element and encapsulate the hacky implementation.

Feel free to tackle it if you're interested. Otherwise, I don't think it's worth the trouble.

When we started this project, I had high ambitions for creating a web UI that rivaled native UI. But I've come to realize that without a higher-level framework that encapsulates all the implementation details of UI animations, interactions, gestures, etc, building a web UI that rivaled native UI is not going to be possible for us.

I think we need to be judicious about how we design our UI UX so that we don't run into situations like this.

**Comment by forabi** [comment]
I experimented with the same trick Google uses for their mobile homepage and it worked. The problem is that it requires a fixed height on the scroll container, which means we have to know how many pixels the pictures and the names are going to take vertically before we are able to hide the scrollbar. This is possible but it requires two render passes: the first one to get the height and the second one to hide the scrollbar, also the height could change if the screen orientation changes... etc. so we need to _monitor_ changes to the component size. This is also possible with [`ResizeObserver`](https://developers.google.com/web/updates/2016/10/resizeobserver) (which [Safari does not support currently](https://caniuse.com/#feat=resizeobserver), which means we would have to use a poll-based polyfill).

**Comment by wholesomedev** [comment]
Holy cow!

**Comment by forabi** [comment]
I guess it's simpler in Google's homepage because they don't have to account for all of this?

**Comment by wholesomedev** [comment]
@forabi Perhaps, I don't know if they have some UX that allows those component sizes to change?

But even in our case, I don't think having elastic Related People component sizes is critical.

(Side thought: one way to slightly improve it is perhaps to add some padding/margin to the bottom so that the scrollbar will not be touching the components but will have its own track)
